src/clj/chronograph/domain/organization.clj

```diff
@@ -1,33 +1,33 @@
 (ns chronograph.domain.organization
-  (:require [chronograph.domain.task :as task]
-            [chronograph.db.organization :as db-organization]
+  (:require [chronograph.db.organization :as db-organization]
             [chronograph.db.user :as db-user]
             [chronograph.domain.acl :as acl]))
 
 (defn create! [tx organization owner-id]
-  (let [{:keys [organizations/id] :as organization} (db-organization/create! tx organization)]
-    (acl/create! tx {:user-id owner-id
-                     :organization-id id
-                     :role acl/admin})
+  (let [{:keys [organizations/id] :as organization}
+        (db-organization/create! tx (select-keys organization
+                                                 [:organizations/name
+                                                  :organizations/slug]))]
+    (acl/create! tx {:acls/user-id owner-id
+                     :acls/organization-id id
+                     :acls/role acl/admin})
     organization))
 
 (defn find-if-authorized
   [tx slug user-id]
   (when-let [{:organizations/keys [id]
-              :as organization} (db-organization/find-by-slug tx slug)]
-    (when (acl/belongs-to-org? tx
-                               user-id
-                               id)
+              :as organization} (db-organization/find-by
+                                 tx
+                                 {:organizations/slug slug})]
+    (when (acl/belongs-to-org? tx user-id id)
```

Comment:
Why not check the acl belonging-ness first and then do the find-by query?

```clojure
(when (acl....)
  (db-organizations/find-by tx {}))
```

Reply:
This was the existing logic.